include/shell/shell.h

```diff
@@ -159,7 +148,32 @@ struct shell_static_entry {
 			STRINGIFY(UTIL_CAT(shell_root_cmd_, syntax)))))	   \
 	__attribute__((used)) = {					   \
 		.is_dynamic = false,					   \
-		.u.entry = &UTIL_CAT(_shell_, syntax)			   \
+		.u = { .entry = &UTIL_CAT(_shell_, syntax) }		   \
+	}
+
+/**
+ * @brief Macro for creating a subcommand set. It must be used outside of any
+ * function body.
+ *
+ * Example usage:
+ * SHELL_STATIC_SUBCMD_SET_CREATE(
+ *	foo,
+ *	SHELL_CMD(abc, ...),
+ *	SHELL_CMD(def, ...)
+ * )
+ *
+ * @param[in] name	Name of the subcommand set.
+ * @param[in] ...	List of commands created with @ref SHELL_CMD_ARG or
+ *			or @ref SHELL_CMD
+ */
+#define SHELL_STATIC_SUBCMD_SET_CREATE(name, ...)			\
+	static const struct shell_static_entry shell_##name[] = {	\
+		__VA_ARGS__,						\
+		SHELL_SUBCMD_SET_END					\
+	};								\
+	static const struct shell_cmd_entry name = {			\
```

Comment:
Do we consider removing `static` here? Variable name could be more complex like: `shell_root_cmd_##name`

Reply:
right, i think if i add prefix we could remove static.